src/app/shared/palette/complex/read-complex-field.component.ts

```diff
@@ -1,9 +1,30 @@
 import { Component } from '@angular/core';
 import { AbstractFieldReadComponent } from '../base-field/abstract-field-read.component';
+import { CaseField } from '../../domain/definition/case-field.model';
 
 @Component({
   selector: 'ccd-read-complex-type-field',
   templateUrl: './read-complex-field.html',
   styleUrls: ['./read-complex-field.scss']
 })
-export class ReadComplexFieldComponent extends AbstractFieldReadComponent {}
+export class ReadComplexFieldComponent extends AbstractFieldReadComponent {
+
+  getComplexFields() {
+    let complexFields = [];
+
+    this.caseField.field_type.complex_fields.forEach((elem, index) => {
+      let elemCopy = this.cloneObject(elem);
+      if (this.caseField.value) {
+        elemCopy.value = this.caseField.value[elem.id];
+        if (!elem.value) {
+          complexFields[index] = elemCopy;
+        }
+      }
+    });
+    return complexFields;
+  }
+
+  private cloneObject(obj: any): any {
```

Comment:
we have in field.utils.ts the same method. At the moment is private. Shall we make it public and reuse that one?